src/TaskLineRenderer.ts

```diff
@@ -240,7 +265,7 @@ async function taskToHtml(
     // So if the priority was not rendered, force it through the pipe of getting the component data for the
     // priority field.
     if (allAttributes.taskPriority === undefined) {
-        const priorityDataAttribute = getComponentDataAttribute('priority', task);
+        const priorityDataAttribute = FieldLayouts['priority'].getDataAttribute('priority', task);
```

Comment:
I'm hugely sorry to do a bit of a U-turn, but I have learned two related things

1. I looked at the title of this PR - _refactor: move TaskLineRender functions to TaskLayoutDetails_  - as a last check before merging, because i couldn't see how `TaskLayoutDetails` related to the change. 
    - I think it probably meant `FieldLayoutDetail`
2. Now that I have resolved all the comments, I see a pattern in the code that I had not noticed before...

I now see that this change inlines `getComponentDataAttribute()` rather than moving it. This turns out to be the cause of the duplicated references to the component name, and the need to export `FieldLayouts`.

May I suggest that this PR is adjusted so that `getComponentDataAttribute()` becomes a static method on `FieldLayoutDetail`, as the PR name suggests - and then that is perfectly good enough for now.

Sorry for the U-turn, but I do think it's better not to inline an existing function instead of moving it, and that change would be more easily reverted in this PR than a separate one.

Reply:
Don't worry, I've made the change and will push it...

Reply:
> May I suggest that this PR is adjusted so that `getComponentDataAttribute()` becomes a static method on `FieldLayoutDetail`, as the PR name suggests - and then that is perfectly good enough for now.

This seems like a nice idea, but aren't static methods in general a smell?

I already wrote the code for the next PR: https://github.com/ilandikov/obsidian-tasks/compare/refactor-tlr-methods-to-fld...ilandikov:obsidian-tasks:refactor-fld-container

It that seems closer to what you wish to see in the code, I propose to merge this PR and continue with the next one. Or we can go the "static" way. Up to you.